oak_runtime/src/node/http/mod.rs

```diff
@@ -444,35 +460,28 @@ fn get_oak_label(req: &HttpRequest) -> Result<Label, OakStatus> {
         (Some(json_label), None) => parse_json_label(json_label.to_vec()),
         (None, Some(protobuf_label)) => parse_protobuf_label(&protobuf_label[..]),
         _ => {
-            warn!(
+            let error_msg = format!(
                 "Exactly one header must be provided as an {} or {} header.",
                 oak_abi::OAK_LABEL_HTTP_JSON_KEY,
                 oak_abi::OAK_LABEL_HTTP_PROTOBUF_KEY
             );
-            Err(OakStatus::ErrInvalidArgs)
+            Err(error_msg)
         }
     }
 }
 
-fn parse_json_label(label_str: Vec<u8>) -> Result<Label, OakStatus> {
+fn parse_json_label(label_str: Vec<u8>) -> Result<Label, String> {
     let label_str = String::from_utf8(label_str).map_err(|err| {
-        warn!(
+        format!(
             "The label must be a valid UTF-8 JSON-formatted string: {}",
             err
-        );
-        OakStatus::ErrInvalidArgs
+        )
     })?;
-    serde_json::from_str(&label_str).map_err(|err| {
-        warn!("Could not parse HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
-    })
+    serde_json::from_str(&label_str).map_err(|err| format!("Could not parse HTTP label: {}", err))
 }
 
-fn parse_protobuf_label(protobuf_label: &[u8]) -> Result<Label, OakStatus> {
-    Label::decode(protobuf_label).map_err(|err| {
-        warn!("Could not parse HTTP label: {}", err);
-        OakStatus::ErrInvalidArgs
-    })
+fn parse_protobuf_label(protobuf_label: &[u8]) -> Result<Label, String> {
+    Label::decode(protobuf_label).map_err(|err| format!("Could not parse HTTP label: {}", err))
```

Comment:
I think we need to be careful about anything that results in a user-facing message, for instance AFAICT now all these strings may end up being displayed to a user in a browser because we use them as errors in the various Results. A better practice is instead to log the error, and return a more fixed (and explicitly whitelisted) message to the user, which actually maps quite nicely to the usual pattern in Rust of defining an enum for the possible errors, and then having a way of converting that enum to a message in the function in which all the errors are handled.

In our case, I think we should probably be even more cautious, so I think it's fine to log the precise error on the server, and return a more generic error to the client, without risking revealing too much information to the user.

Reply:
I agree. How about just responding with a generic response like "Invalid or missing oak-label" in line 332?

Reply:
Yeah I think that's the safest for now, and as long as we log the more precise error, we should be able to debug things if anything goes wrong, at least while we have logging enabled.

But I would still change all the errors, it is not very idiomatic to use strings as errors. If you want unstructured errors, then perhaps anyhow could be used, but we don't normally use it at the library level, only at top level for binaries.

If there is no enum that does the job here, you could define a new one, or just use () as the error type, or use Option instead of Result.

Reply:
Done.